framework/include/mesh/FileMesh.h

```diff
@@ -28,14 +29,13 @@ class FileMesh : public MooseMesh
 public:
   FileMesh(const InputParameters & parameters);
   FileMesh(const FileMesh & other_mesh);
-  virtual ~FileMesh();
```

Comment:
Same goes for this.  If we declare `~FileMesh` as empty in the C file, we should not need to `#include "libmesh/exodusII_io.h"` here.  It's a little more verbose, but I think it's preferable to reintroducing all these header file dependencies that I worked so hard to get rid of.